src/index.ts

```diff
@@ -40,78 +29,72 @@ async function bundleMDX(
   const absoluteFiles: Record<string, string> = {
     [entryPath]: entryCode,
   }
   for (const [filepath, fileCode] of Object.entries(files)) {
     absoluteFiles[path.join(dir, filepath)] = fileCode
   }
-  const inMemoryModulePlugin = {
-    name: 'in-memory-module',
-    resolveId(importee: string, importer?: string) {
-      if (!importer || importee === entryPath) return importee
-      if (!importee[0].startsWith('.')) return null
-
-      const resolved = path.resolve(path.dirname(importer), importee)
-
-      if (resolved in absoluteFiles) return resolved
-      for (const ext of ['.js', '.tsx', '.jsx', '.ts']) {
-        const resolvedWithExt = `${resolved}${ext}`
-        if (resolvedWithExt in absoluteFiles) return resolvedWithExt
-      }
-      throw new Error(
-        `Could not resolve '${importee}' from ${
-          importer === entryPath
-            ? 'the entry MDX file'
-            : `'${importer.replace(`${dir}/`, '')}'`
-        }`,
-      )
-    },
-    load(id: string) {
-      if (absoluteFiles[id]) return absoluteFiles[id]
-      return null
+
+  const inMemoryPlugin: Plugin = {
+    name: 'inMemory',
+    setup(build) {
+      build.onResolve({filter: /.*/}, args => {
+        if (args.path === entryPath) return {path: args.path}
+
+        const modulePath = path.resolve(path.dirname(args.importer), args.path)
+
+        if (modulePath in absoluteFiles) return {path: modulePath}
+
+        for (const ext of ['.js', '.ts', '.jsx', '.tsx']) {
+          const fullModulePath = `${modulePath}${ext}`
+          if (fullModulePath in absoluteFiles) return {path: fullModulePath}
+        }
+
+        console.dir(args.importer)
+
+        throw new Error(
+          `Could not resolve ${args.path} in ${
+            args.importer === entryPath
+              ? 'the entry MDX file.'
+              : args.importer.replace(dir, '.')
+          }`,
+        )
+      })
+
+      build.onLoad({filter: /__mdx_bundler_fake_dir__/}, args => {
+        return {
+          contents: absoluteFiles[args.path],
+          loader: 'jsx',
+        }
+      })
     },
   }
 
-  const inputOptions = getInputOptions({
-    external: ['react', 'react-dom'],
-    input: entryPath,
+  const buildOptions = esbuild({
+    entryPoints: [entryPath],
+    write: false,
     plugins: [
-      inMemoryModulePlugin,
-      nodeResolve(),
-      commonjs({include: 'node_modules/**', sourceMap: false}),
-      json(),
-      rollupBabel({
-        babelHelpers: 'inline',
-        configFile: false,
-        exclude: /node_modules/,
-        extensions: ['.js', '.ts', '.tsx', '.md', '.mdx', '.jsx', '.json'],
-        presets: [
-          [babelPresetReact, {pragma: 'mdx'}],
-          babelPresetEnv,
-          [babelPresetTypeScript, {allExtensions: true, isTSX: true}],
-        ],
-        sourceMaps: false,
+      nodeResolve({
+        extensions: ['.js', '.ts', '.jsx', '.tsx'],
       }),
-      terser(),
```

Comment:
Does esbuild minify automatically?

Reply:
I don’t think so

Reply:
There is a `minify` option in esbuild which should do the trick.

Reply:
I've not got esbuild to minify the output.